torch/cuda/amp/autocast_mode.py

```diff
@@ -13,6 +13,11 @@ class autocast(torch.amp.autocast_mode.autocast):
     ``torch.cuda.amp.autocast(args...)`` is deprecated. Please use ``torch.amp.autocast("cuda", args...)`` instead.
     """
 
+    @deprecated(
+        "`torch.cuda.amp.autocast(args...)` is deprecated. "
+        "Please use `torch.amp.autocast('cuda', args...)` instead.",
+        category=FutureWarning,
+    )
```

Comment:
Decorating `DeprecatedClass.__init__` rather than the `DeprecatedClass` itself here due to an upstream issue.

- python/cpython#119605

`torch` JIT uses the `inspect` module to get the type annotations of custom methods of a class. The `@deprecated()` decorator will generated `__new__` and `__init_subclass__` if not defined by users. This behavior is not compatible with `inspect.getsourcelines` and `inspect.signature`.